jsettlers.graphics/src/jsettlers/graphics/utils/UIList.java

```diff
@@ -49,19 +49,18 @@
 		UIListItem getItem(T item);
 	}
 
-	public UIList(List<? extends T> items, ListItemGenerator<T> generator,
-			float itemheight) {
-		this.items = items;
+	public UIList(List<? extends T> items, ListItemGenerator<T> generator, float itemheight) {
+		setItems(items);
 		this.generator = generator;
 		this.itemheight = itemheight;
-		if (items.size() > 0) {
-			activeItem = items.get(0);
-		}
 	}
 
 	public void setItems(List<? extends T> list) {
 		synchronized (itemsMutex) {
 			this.items = list;
+			if (items.size() > 0) {
```

Comment:
If a new set of items is set for the UIList then the previous active item would be invalid. In fact when the new list size is 0 the active item should be set to null. There should also really be a check on the setActiveItem method as well to check it's in the list.
Additions would be made on the List object (that's passed in) and these will be reflected in the UI without changing the user's selection.